lib/TokenManager.js

```diff
@@ -24,51 +24,46 @@ function emitExpired(tokenMgmtRef, key, token) {
   tokenMgmtRef.emitter.emit('expired', key, token);
 }
 
-function clearRefreshTimeout(tokenMgmtRef, key) {
-  clearTimeout(tokenMgmtRef.refreshTimeouts[key]);
-  delete tokenMgmtRef.refreshTimeouts[key];
+function emitError(tokenMgmtRef, error) {
+  tokenMgmtRef.emitter.emit('error', error);
 }
 
-function clearRefreshTimeoutAll(tokenMgmtRef) {
-  var refreshTimeouts = tokenMgmtRef.refreshTimeouts;
-  for(var key in refreshTimeouts) {
-    if (!refreshTimeouts.hasOwnProperty(key)) {
+function clearExpireEventTimeout(tokenMgmtRef, key) {
+  clearTimeout(tokenMgmtRef.expireTimeouts[key]);
+  delete tokenMgmtRef.expireTimeouts[key];
+}
+
+function clearExpireEventTimeoutAll(tokenMgmtRef) {
+  var expireTimeouts = tokenMgmtRef.expireTimeouts;
+  for (var key in expireTimeouts) {
+    if (!expireTimeouts.hasOwnProperty(key)) {
       continue;
     }
-    clearRefreshTimeout(tokenMgmtRef, key);
+    clearExpireEventTimeout(tokenMgmtRef, key);
   }
-  tokenMgmtRef.refreshTimeouts = {};
 }
 
-function setRefreshTimeout(sdk, tokenMgmtRef, storage, key, token) {
-  var refreshWait = (token.expiresAt * 1000) - Date.now();
-  if (refreshWait < 0) {
-    // Already expired
-    refreshWait = 0;
-  }
-  var refreshTimeout = setTimeout(function() {
-    if (tokenMgmtRef.autoRefresh) {
-      return refresh(sdk, tokenMgmtRef, storage, key);
-    } else if (token.expiresAt * 1000 <= Date.now()) {
-      remove(tokenMgmtRef, storage, key);
-      emitExpired(tokenMgmtRef, key, token);
-    }
-  }, refreshWait);
+function setExpireEventTimeout(tokenMgmtRef, key, token) {
+  var expireEventWait = (token.expiresAt * 1000) - Date.now();
+
+  var expireEventTimeout = setTimeout(function() {
+    emitExpired(tokenMgmtRef, key, token);
+  }, expireEventWait);
 
   // Clear any existing timeout
-  clearRefreshTimeout(tokenMgmtRef, key);
+  clearExpireEventTimeout(tokenMgmtRef, key);
```

Comment:
nit: Would be a little nicer to put this before the previous line - logic is clear old timeout, create new timeout, set reference for new timeout.